main.tf

```diff
@@ -52,6 +52,27 @@ resource "ibm_resource_instance" "secrets_manager_instance" {
   }
 }
 
+# Create IAM credentials engine using s2s auth
+resource "ibm_iam_authorization_policy" "iam_identity_policy" {
+  count                       = var.create_iam_engine ? 1 : 0
+  source_service_name         = "secrets-manager"
+  source_resource_instance_id = local.secrets_manager_guid
+  target_service_name         = "iam-identity"
+  roles                       = ["Operator"]
+  description                 = "Authorization Policy"
```

Comment:
The description needs more info as to the type of auth policy here. Check out other auth policy descrptions we have used in other modules for an idea